src/Symfony/Component/Translation/Bridge/Crowdin/CrowdinProvider.php

```diff
@@ -0,0 +1,397 @@
+        $response = $this->client->request('PUT', sprintf('https://%s/projects/%s/files/%d', $this->endpoint, $this->projectId, $fileId), [
+            'headers' => [
+                'Content-Type' => 'application/json',
+            ],
+            'body' => json_encode([
+                'storageId' => $storageId,
+            ]),
+        ]);
+
+        if (200 !== $response->getStatusCode()) {
+            throw new ProviderException(sprintf('Unable to update file in Crowdin for file ID "%d" and domain "%s".', $fileId, $domain), $response);
+        }
+    }
+
+    private function uploadTranslations(int $fileId, string $domain, string $content, string $locale): void
+    {
+        if (!$fileId) {
+            return;
+        }
+
+        $storageId = $this->addStorage($domain, $content);
+
+        /**
+         * @see https://support.crowdin.com/api/v2/#operation/api.projects.translations.postOnLanguage (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.projects.translations.postOnLanguage (Crowdin Enterprise API)
+         */
+        $response = $this->client->request('POST', sprintf('https://%s/projects/%s/translations/%s', $this->endpoint, $this->projectId, $locale), [
+            'headers' => [
+                'Content-Type' => 'application/json',
+            ],
+            'body' => json_encode([
+                'storageId' => $storageId,
+                'fileId' => $fileId,
+            ]),
+        ]);
+
+        if (200 !== $response->getStatusCode()) {
+            throw new ProviderException(sprintf('Unable to upload translations to Crowdin for domain "%s" and locale "%s".', $domain, $locale), $response);
+        }
+    }
+
+    private function exportProjectTranslations(string $languageId, int $fileId): string
+    {
+        /**
+         * @see https://support.crowdin.com/api/v2/#operation/api.projects.translations.exports.post (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.projects.translations.exports.post (Crowdin Enterprise API)
+         */
+        $response = $this->client->request('POST', sprintf('https://%s/projects/%d/translations/exports', $this->endpoint, $this->projectId), [
+            'headers' => [
+                'Content-Type' => 'application/json',
+            ],
+            'body' => json_encode([
+                'targetLanguageId' => $languageId,
+                'fileIds' => [$fileId],
+            ]),
+        ]);
+
+        if (204 === $response->getStatusCode()) {
+            throw new ProviderException(sprintf('No content in exported file %d.', $fileId), $response);
+        }
+
+        if (200 !== $response->getStatusCode()) {
+            throw new ProviderException(sprintf('Unable to export file %d translations for language "%s".', $fileId, $languageId), $response);
+        }
+
+        $export = json_decode($response->getContent(), true);
+
+        return file_get_contents($export['data']['url']);
```

Comment:
use the HTTP client for that

Reply:
I tried to use the HTTP client passed to this Provider, but it already configured with Crowdin base URI and Authorization header and it caused the Bad Request error because the URL from the response is not a Crowdin link, it's an S3 link

I didn't found a way to reset the base URI and Authorization header. Should I create a new instance of HTTP client to make this request?

Reply:
No. I think you should rather take a generic HttpClient in the constructor rather than a scoped one if you need to do more HTTP requests